ci/pipelines/deploy-to-staging.yml

```diff
@@ -460,7 +477,82 @@ jobs:
         trigger: true
       - set_pipeline: deploy-to-staging
         file: deploy-to-staging-pipeline-definition/ci/pipelines/deploy-to-staging.yml
-        
+
+  - name: deploy-carbon-relay
+    plan:
+      - get: carbon-relay-ecr-registry-staging
+        trigger: true
+      - get: pay-infra
+      - get: pay-ci
+      - load_var: carbon_relay_image_tag
+        file: carbon-relay-ecr-registry-staging/tag
+      - task: create-failure-notification-snippet
+        config:
+          platform: linux
+          image_resource:
+            type: docker-image
+            source:
+              repository: alpine
+          params:
+            CARBON_RELAY_IMAGE_TAG: ((.:carbon_relay_image_tag))
+            ENV: staging-2
+          outputs:
+            - name: snippet  
+          run:
+            path: sh
+            args:
+              - -c
+              - |
+                cat <<EOT >> snippet/failure
+                :red_circle: FARGATE Deployment of <https://github.com/alphagov/pay-dockerfiles/releases/tag/${CARBON_RELAY_IMAGE_TAG}|carbon-relay v${CARBON_RELAY_IMAGE_TAG}> on ${ENV} failed. Version details:
+                EOT
+      - load_var: failure_snippet
+        file: snippet/failure
```

Comment:
For staging app releases we usually have a success notification as well. I'm happy for this to be added in a subsequent PR though.

Reply:
I thought we agreed that we would be happy for just a failure notification for the carbon-relay deploy? That's the way it is for the deploy-to-test. From https://github.com/alphagov/pay-ci/pull/500:
```
The team are in agreement that just like for the
nginx/nginx-forward-proxy/telegraf deploys, we only need failure notifications
for carbon-relay deploys.
```
It is an app and not a sidecar like nginx is though, so I can see the rationale for the start and success notifications as well. Happy to change it if you like?

Reply:
If we want a start and success notification it'd be easier to do in another PR.

Reply:
Hmm good point! I think the reasoning behind the auto-slack notifications is so that devs don't have to post about the release in the channel themselves. The telegraf/nginx stuff only ever goes out with an app, so we'd 'know' about it via that app's notification. Whereas the carbon relay app is its own task (I don't think we'd need separate notifications for stunnel, for example, as that's always going out with a carbon relay 'app' release). Maybe we should add start/success to the test pipeline after all, and send them to the `govuk-pay-activity` black hole?

Reply:
(To reiterate, I don't think we need to do it in this PR)